preset.module

```diff
@@ -341,3 +374,50 @@ function preset_get($module, $preset_type = NULL) {
     return $preset_types;
   }
 }
+
+/**
+ * Gets configuration for all presets.
+ * 
+ * @param string $module
+ *   The module name to get preset types from.
+ * @param string $preset_type
+ *   Get a specific preset type. Set to `NULL` to get all preset types for the
+ *   given module.
+ * @param array $properties
+ *   Properties of the preset type.
+ * @param string $id
+ *   The mahcine name of a specific preset.
+ *
+ * @return array
+ *   An associative array of all preset configurations keyed by machine name.
+ */
+function preset_get_config($module, $preset_type, $properties = array(), $id = FALSE) {
+  if (empty($properties)) {
+    $properties = preset_get($module, $preset_type);
+  }
+
+  if (array_key_exists('individual', $properties) && $properties['individual']) {
+    // Multple config files
+    $config_names = config_get_names_with_prefix($module . '.' . $preset_type);
+    $configs = array();
+    foreach($config_names as $name) {
+      $name_parts = explode('.', $name);
+      $id = array_pop($name_parts);
```

Comment:
This use of `$id` conflicts with the variable passed in the function call -- maybe we can change this usage to `$config_id`.